kroxylicious-operator/src/main/resources/META-INF/fabric8/record-encryptions.filter.kroxylicious.io-v1.yml

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+# Note the v1 in the filename refers to the version of the CustomResourceDefinition
+# not any of the versions of API being defined.
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  # name must match the spec fields below, and be in the form: <plural>.<group>
+  name: record-encryptions.filter.kroxylicious.io
+spec:
+  # group name to use for REST API: /apis/<group>/<version>
+  group: filter.kroxylicious.io
+  scope: Namespaced
+  names:
+    categories:
+      - kroxylicious-plugins
+    plural: record-encryptions
+    singular: record-encryption
+    kind: RecordEncryption
+    shortNames:
+      - kre
+  # list of versions supported by this CustomResourceDefinition
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            metadata:
+              type: object
+              properties:
+                name:
+                  type: string
+                  maxLength: 63
+                  pattern: "[a-z0-9]([a-z0-9-]*[a-z0-9])?"
+            spec:
+              type: object
+              properties:
+                kms:
+                  type: string
+                kmsConfig:
+                  type: object
+                selector:
```

Comment:
Don't the `*Config` objects needs:

`x-kubernetes-preserve-unknown-fields: true`

in order to carry the type specific configuration down to the config.